backend/src/main/java/com/htecgroup/skynest/util/UrlUtil.java

```diff
@@ -2,9 +2,13 @@
 
 import lombok.AccessLevel;
 import lombok.NoArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
 import java.util.stream.Stream;
 
+@Component
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class UrlUtil {
```

Comment:
good practice is if the constructor is private to mark the class as final, meaning it is closed for extensions.